### PR TITLE
Vox nukies and ninjafix

### PR DESCRIPTION
**Pull request by Errant-4**
## About the PR
Mobs spawned from AntagselectionSystem can now use Loadouts (in practice, currently this is only relevant to nukies)
As this fulfills the requirements stated for Vox to go back to nukies (#29707), I have removed them from the blacklist

edit: Space Ninjas were just changed to use AntagSelectionSystem, and are now also updated for species loadouts

## Technical Details
AntagselectionSystem now uses LoadoutsSystem's gear spawn code, rather than calling SharedStationSpawningSystem directly. This makes ghostrole gear spawning work consistently, and does not require this code to be updated in multiple places when it changes in the future

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Errant
- tweak: Vox can be nukies and ninjas again.

### CODE REVIEW

**Comment by beck-thompson** [comment]
You are the vox hero!

**Comment by Errant-4** [comment]
Hold up, I was quite sleepy last night and did not think to kill the unnecessary amount of pain resulting from having these be different code paths in the first place. I will make AntagSelectionSystem call LoadOutSystem for the equipping

File: Content.Server/Antag/Components/AntagSelectionComponent.cs [hunk 1]
```diff
@@ -154,6 +155,12 @@ public partial struct AntagSelectionDefinition()
     [DataField]
     public ProtoId<StartingGearPrototype>? StartingGear;
 
+    /// <summary>
+    /// A list of role loadouts to be equipped to the player.
+    /// </summary>
+    [DataField]
+    public List<ProtoId<RoleLoadoutPrototype>>? RoleLoadout;
+
```
Comment (metalgearsloth) [inline Content.Server/Antag/Components/AntagSelectionComponent.cs:163]: Is there even a reason to have startinggear anymore at that point considering they can contain multiple startinggears?

Also having a list doesn't make sense here unless it's randomly selected.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Antag/Components/AntagSelectionComponent.cs (hunk 1 above)
Comment (Errant-4) [reply Content.Server/Antag/Components/AntagSelectionComponent.cs:163]: Strictly speaking, no we don't need startinggear on this component anymore. We could put the existing startinggear prototypes inside loadoutgroups. Downstreams might still need startinggear, for all I know. Do you want me to remove startinggear from this component and just put it in breaking changes?

There is in fact a random selection for the roleloadout list, I just commented it poorly. I will change that now

Comment (Errant-4) [reply Content.Server/Antag/Components/AntagSelectionComponent.cs]: I did the thing. I will phase out using startingGear on a separate PR as its kind of a big delta

File: Content.Shared/Clothing/LoadoutSystem.cs [hunk 2]
```diff
@@ -139,22 +139,38 @@ public string GetName(IEquipmentLoadout? gear)
...
         var loadout = new RoleLoadout(id);
         loadout.SetDefault(GetProfile(uid), _actors.GetSession(uid), _protoMan, true);
         _station.EquipRoleLoadout(uid, loadout, proto);
+
+        GearEquipped(uid);
+    }
+
+    public void GearEquipped(EntityUid uid)
+    {
+        // Internals try to auto-activate on this event
```
Comment (metalgearsloth) [inline Content.Shared/Clothing/LoadoutSystem.cs:171]:
```suggestion
    {
```

Guaranteed to go out of date at some point.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**